src/app/authentication/_services/authentication.service.ts

```diff
@@ -193,15 +207,44 @@ export class AuthenticationService {
     }
   }
 
-  public hasRole(role: string): Observable<boolean> {
-    return this.roleSubject.asObservable().pipe(
-      map(roles => {
-        if (roles) {
-          return roles.includes(role);
-        }
-        return null;
-      })
-    );
+  get user(): Observable<UserInfoModel> {
+    return this.userSubject.asObservable();
+  }
+
+  get roles(): Observable<UserRole[]> {
+    return this.rolePASubject.asObservable();
+  }
+
+  /** Authentication Flow Helper */
+  // Generate a secure random string using the browser crypto functions
+  generateRandomString(length: number) {
+    var array = new Uint32Array(length/2); // one byte is translated into two characters => divide by 2
+    window.crypto.getRandomValues(array);
+    return Array.from(array, dec => ('0' + dec.toString(16)).substr(-2)).join('');
```

Comment:
It does convert the binary to hex using toString(16). It should not be able to produce EOF or similar because of that.

The ``'0' + ...`` and ``substr(-2)`` are only for padding in case the current integer is small enough to get converted into a 1 character hex representation.

However, there is a potential bug in case the supplied length is odd (which is currently just not the case): 
if ``length`` = 2n+1, then ``len(array)`` = n which results in a returned string of size 2n and not 2n+1

Will fix this by always creating a marginally larger array ((length+1)/2) and using substring to return the correct size.